File: ruby/pg-test/fiber-bench.rb
require 'benchmark'
require 'fiber'

f1=Fiber.new do
  while true
    Fiber.yield
  end
end

f2=Fiber.new do
  while true
    Fiber.yield
  end
end

COUNT=50000000

Benchmark.bm do |x|
  x.report{
    COUNT.times do
      f1.resume
      f2.resume
    end
  }
end

File: ruby/pg-test/fiber-bench.lua
c1=coroutine.create(function ()
                     while true do
                       coroutine.yield()
                     end
                    end)

c2=coroutine.create(function ()
                     while true do
                       coroutine.yield()
                     end
                    end)

local start=os.clock()
local count=50000000

for i=1,count do
 coroutine.resume(c1)
 coroutine.resume(c2)
end

print(4*count/(os.clock()-start))
